### PR TITLE
in cronvar, unsetting a default of 'root'. a default of root prevents…

### PR DESCRIPTION
**Pull request by zahna**
… this module from working with a non-root remote user.

This is pretty much it. Removing the 2 lines that sets the default user to root enables this module to work with a non-root remote user, while keeping correct behavior with the root remote user.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @zahna. @dougluce please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: system/cronvar.py [hunk 1]
```diff
@@ -114,14 +114,12 @@ class CronVar(object):
     """
         CronVar object to write variables to crontabs.
 
-        user      - the user of the crontab (defaults to root)
+        user      - the user of the crontab (defaults to None)
```
Comment (dougluce) [inline system/cronvar.py]: Should this be "defaults to ansible_user" instead of "defaults to None?"

Comment (zahna) [inline system/cronvar.py]: Well, the value of self.user = user, which I set to have a placeholder value of None.

Comment (allenluce) [inline system/cronvar.py]: In which case it defaults to the ansible_user.  It doesn't default to a literal "None" nor does it default to no user. Change this to "defaults to ansible_user" and I'll be fine with the PR.

Comment (zahna) [inline system/cronvar.py]: Done. I also corrected one other line for comment consistency.

**Comment by gregdek** [comment]
@dougluce This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by dougluce** [comment]
I might update the docs about the default to something like "the default ansible user" (or whatever you call the username that's running the scripts on the target host). But in general I'm ok with this -- shipit.

**Comment by gregdek** [comment]
Thanks again to @zahna for this PR, and thanks @dougluce for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
hmm, this would break backwards compatibility right? 

needs_info

**Comment by gregdek** [comment]
Thanks @zahna for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@zahna A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@zahna Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by zahna** [comment]
Hi, Greg. Thanks for the reminder. Is my patch breaking the Travis build? I'm not sure what resmo is asking above. Does it break backwards compatibility? Technically, I suppose it can, but the previous behavior was incorrect. When setting a cronvar, when one doesn't specify a user, one would expect that the cronvar would be set in the crontab of the ansible remote_user, which many times is root, but not always. Setting a default of 'root' works when the remote_user is root, but breaks when the remote_user is non-root. This change corrects this behavior.  
Addendum: I did double check specified behavior in the docs, and the docs do list the default user for cronvar being "root". However, I still do believe that the best behavior is to default to the ansible remote_user.  http://docs.ansible.com/ansible/cronvar_module.html

**Comment by gregdek** [comment]
@zahna A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by zahna** [comment]
I'm not sure what else I'm supposed to do here. More info was requested and I provided it. I still think the PR is ready_for_review.

**Comment by dougluce** [comment]
Looking at this again, I wonder if the default value should be "null" (i.e. no default) in the docs. Have you tested this with that ansible_user setting?

**Comment by zahna** [comment]
I made this change a while ago, but from what I remember the cronvar module
complained in the case of user not having a value. However, in looking at
my ansible plays, I have noticed that my change to cronvar bring cronvar's
behavior in line with cron's. I know the docs say that cron specifies a
default user of "root", but that isn't accurate. With my version of
cronvar.py, the following tasks are both enacted in the remote_user's
crontab:

```
- cronvar: name="MAILTO" value="<email_snipped>" state="present"
- cron: name="some_script"
job="{{site['docroot']}}/shared/scripts/somescript.sh" hour="0" minute="0"
state="present"
```

EDIT: I just realized that I never explicitly answered your question. I have tested my change in that I have been using my modified version of cronvar in production. It works as expected.

**Comment by gregdek** [comment]
OK, we may be running into an age-related Travis issue at this point. @zahna if you could rebase this PR to latest devel, we can get this thing moving again. Thanks!

**Comment by gregdek** [comment]
Thanks @zahna for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@zahna A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by zahna** [comment]
I have rebased. ready_for_review.

**Comment by dougluce** [comment]
@zahna Looks like you merged instead of rebasing. Can you do this in your branch?
- `git reset --hard 156db17`
- `git rebase upstream/devel` (or whatever your remote ref to this repo is)
- `git push -f`

**Comment by gregdek** [comment]
Thanks @zahna for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by zahna** [comment]
That's odd. I used "git rebase ..." after adding the new remote. Anyway, I
performed another rebase per your instructions. let me know if there's
anything else I need to do.

**Pull request closed without merging**